desktop/cmp/filechooser/FileChooserModel.ts

```diff
@@ -113,19 +173,47 @@ export class FileChooserModel extends HoistModel {
     }
 
     @action
-    onDrop(accepted, rejected, enableMulti) {
-        if (!isEmpty(accepted)) {
-            if (!enableMulti) {
-                this.setSingleFile(accepted[0]);
-            } else {
-                this.addFiles(accepted);
-            }
+    onDrop(accepted: File[], rejected: FileRejection[]) {
+        if (accepted?.length > 1 && !this.enableMulti) {
+            accepted = [accepted[0]];
         }
-        this.lastRejectedCount = rejected.length;
+        this.lastAccepted = accepted;
+        this.lastRejected = rejected;
+        this.addFiles(accepted);
     }
 
-    onFilesChange(files) {
+    onFilesChange(files: File[]) {
         const fileData = files.map(file => ({name: file.name, size: file.size}));
         this.gridModel.loadData(fileData);
     }
+
+    //----------------
+    // Implementation
+    //----------------
+    private defaultTargetDisplay = () => {
+        return fragment(
+            p('Drag and drop your files here, or click to browse.'),
+            p({
+                items: [
+                    'Note that this example is configured to accept only ',
+                    code(
+                        isArray(this.accept)
+                            ? this.accept.map(type => '*' + type).join(' and ')
+                            : this.accept
+                    ),
+                    ' file types.'
+                ]
+            })
+        );
+    };
+
+    private defaultRejectDisplay = () => {
+        if (isEmpty(this.lastRejected)) return null;
+        console.log('File rejections:', this.lastRejected);
```

Comment:
logging still here

Reply:
I left this here so that the user can access the FileRejection objects for each rejection in the console if they want more details on the reason that the file failed to upload. It was too clunky to keep the reasons in the FileChooser display, as discussed, but it might be helpful to utilize the upgraded Dropzone feature of citing the reason for the upload failure. If not, I'll remove it.